.env.example

```diff
@@ -0,0 +1,29 @@
+###################################################################
+# App parameters
+# To override default configurations, save a copy of this file to 
+#   ~/.freqtrade/.env or
+#   user_data/.env or
+#   user_data/user_repo/.env
+#
+# Just be sure not to set the same variables in duplicate locations.
+# For example, you may want to save secret keys in ~/.freqtrade/.env
+# and other settings to user_data/user_repo/.env
+####################################################################
+# Location of user configuration files. Defaults to ~/.freqtrade
+CONFIG_PATH=
+# No default. EG: git@github.com:me/freqtrade-stuff.git
+# If provided it will be cloned to user_data/user_repo
+USER_REPO=
+# Location of custom strategies. Defaults to user_data/strategies
+STRATEGY_DIR=
+# App Dockerfile. Defaults to Dockerfile
+APP_DOCKERFILE=
+# Default strategy to use with docker commands. Defaults to TestStrategy
+DEFAULT_STRATEGY=
+# Extra arguments to use when starting the app with Docker.
+# Defaults to -s $(DEFAULT_STRATEGY) --strategy-path $(STRATEGY_DIR)
+START_ARGS=
+# Exchange keys in the format EXCHANGE_KEY=123abc
+BINANCE_KEY=
```

Comment:
I don't think these keys are actually read from the bot ... so i guess they are a bit misleading?

Reply:
Yes that is correct. I'll remove them from .env.example. If you like the idea however, I can open a new issue to save exchange credentials in an environment variable.
Ideally, I would like to version control my files, but without my exchange credentials. I would prefer to keep those in ~/.freqtrade rather than in with the code where I might accidentally commit it to the repo.
 
It would be nice to disambiguate the 3 types of configurations to offer more flexibility. 

1. secrets (exchange credentials)
2. bot configurations (conda, venv, docker, rpi)
3. strategy configurations (EG RSI_no_stoploss_1min)

Reply:
> I would prefer to keep those in ~/.freqtrade rather than in with the code where I might accidentally commit it to the repo.

Nice point btw... 👍

Reply:
well you can do that with multiple --config options (the last pointing to `--config  ~/.secret_keys.json` ?  ...

Reply:
I didn't see that in the docs. Like this?

```bash
freqtrade --config user_data/config.json --config ~/.freqtrade/.secret_keys.json --config user_data/user_repo/some_other_config.json
```

I did have trouble setting parameters in the strategies. The first time I messed with the project, I found that some configurations needed to be set in `config.json` even though the docs said they could be set in either place. Is that a known issue or should I try again and debug?

Reply:
I was able to use the multiple config option, so keys can be saved separately from config. The user can populate `START_ARGS` to save the configuration and pass it to Docker. Instead of mounting `config.json`, I mount `$(CONFIG_PATH):user_config`. That provides any config files the user has saved there and the mount of USER_DATA provides access to any config files saved somewhere in that directory.

I made the default `-s $(DEFAULT_STRATEGY) --strategy-path $(STRATEGY_DIR) -c user_config/config.json` which can be overridden in .env. as I have done here:

```bash
# Extra arguments to use when starting the app with Docker.
# Defaults to  -s $(DEFAULT_STRATEGY) --strategy-path $(STRATEGY_DIR) \
#		        -c user_config/config.json
START_ARGS=-c user_config/exchange-config.json \
    -c user_data/user_repo/strategies/new-strategy-config.json \
    -s $(DEFAULT_STRATEGY) --strategy-path $(STRATEGY_DIR)
```

Now I can put secrets in `~/.freqtrade/exchange-config.json` as

```bash
{   "exchange": {
        "name": "binance",
        "key": "your_exchange_key",
        "secret": "your_exchange_secret",
        "ccxt_config": {"enableRateLimit": true},
        "ccxt_async_config": {
            "enableRateLimit": true,
            "rateLimit": 200
        }
    },
    "telegram": {
        "enabled": false,
        "token": "your_telegram_token",
        "chat_id": "your_telegram_chat_id"
    }
}
```

and I can put strategy-type config in version control in my strategies repo as:

```bash
{
    "max_open_trades": 3,
    "stake_currency": "BTC",
    "stake_amount": 0.0005,
    "fiat_display_currency": "USD",
    "ticker_interval" : "1m",
    "exchange": {
        "name": "binance",
        "pair_whitelist": [
            "ETH/BTC",
            "EOS/BTC",
        ],
        "pair_blacklist": [
            "BNB/BTC"
        ]
    },
...

}

```
Thanks for making that possible.

This makes sense to me, but it's getting pretty far from the current installation and configuration strategies. I'll let you decide if this is helpful or just additional complexity that may confuse users. Feel free to quash the idea if I'm out in left field with the Makefile.

Reply:
The makefile makes kindof sense - but it's a rather longish makefile (for a introduction, anyway) - and i wanna make sure we understand it in it's entirely before merging it (which in the end, will mean that we'll need to support it). (That's why reviewing takes a long time).